### ISSUE TITLE
null pointer exception in networkResponse.statusCode

### ISSUE THREAD

**Issue by pradeepkumarreddyk** [swagger-api/swagger-codegen#3628]
The following is the swagger generated code for android:-
-----------------------------------------------------------------
try {
        String localVarResponse = apiInvoker.invokeAPI (basePath, path, "GET", queryParams, postBody, headerParams, formParams, contentType, authNames);
        if(localVarResponse != null){
           return (Lxlresponse) ApiInvoker.deserialize(localVarResponse, "", Lxlresponse.class);
        } else {
           return null;
        }
      } catch (ApiException ex) {
         throw ex;
      } catch (InterruptedException ex) {
         throw ex;
      } catch (ExecutionException ex) {
        
         if(ex.getCause() instanceof VolleyError) {
           
             throw new ApiException(((VolleyError) ex.getCause()).networkResponse.statusCode, ((VolleyError) ex.getCause()).getMessage());
         }
         throw ex;
      } catch (TimeoutException ex) {
         throw ex;
      }
  }

In the above code, at the following line i'm getting null pointer exception, when there is no response from server:-
----------------------------------------------------------------------------------------------------------------
throw new ApiException(((VolleyError) ex.getCause()).networkResponse.statusCode, ((VolleyError) ex.getCause()).getMessage());

**Comment by wing328** [comment]
@pradeepkumarreddyk thanks for reporting the issue. The address it, we need to do a null check for networkResponse before accessing the `statusCode`. If networkResponse is null, we will pass the status code as 0.

May I know if you've cycle to contribute the enhancement?

**Comment by pradeepkumarreddyk** [comment]
i did not understand. what is cycle to contribute the bug fix ?

**Comment by wing328** [comment]
I mean do you have time to contribute the fix.

**Comment by pradeepkumarreddyk** [comment]
yes, i have time to contribute the fix.

**Comment by wing328** [comment]
Yes, you're correct. The following file is a good starting point:

https://github.com/swagger-api/swagger-codegen/blob/1dd9ee39af6c4e2f9c016a94d16ca47a88a9cb0d/modules/swagger-codegen/src/main/resources/android/libraries/volley/api.mustache#L112-L134

**Comment by pradeepkumarreddyk** [comment]
i have cloned the swagger-codegen code, then i have executed the following
command on top level directory to find out the list of files which need
modification.


 *grep -r volleyError.networkResponse **


Output of the above command is as follows :-
----------------------------------------------------------------
modules/swagger-codegen/src/main/resources/android/libraries/volley/api.mustache:
          if (volleyError.networkResponse != null) {
modules/swagger-codegen/src/main/resources/android/libraries/volley/api.mustache:
             throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
    if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/PetApi.java:
       throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
          if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
             throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
          if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
             throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
          if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
             throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
          if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/StoreApi.java:
             throw new ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
           if (volleyError.networkResponse != null) {
samples/client/petstore/android/volley/src/main/java/io/swagger/client/api/UserApi.java:
              throw new
ApiException(volleyError.networkResponse.statusCode,
volleyError.getMessage());


=> i did not find even a single instance where null check is not done.
Everywhere null check is already done in the code.
What should i do ?
